File: src/com/mobiletheatertech/plot/Event.java
package com.mobiletheatertech.plot;

import org.w3c.dom.Element;
import org.w3c.dom.Node;
import org.w3c.dom.NodeList;

import javax.imageio.metadata.IIOMetadataNode;

/**
 * Created with IntelliJ IDEA. User: dhs Date: 11/13/13 Time: 8:00 PM To change this template use
 * File | Settings | File Templates.
 *
 *
 *
 * @author dhs
 * @since 0.0.20
 */
public class Event extends MinderDom {

    private static Event Only = null;

//    public static void ParseXML( NodeList list )
//            throws AttributeMissingException, InvalidXMLException, ReferenceException
//    {
//
//        int length = list.getLength();
//        for (int index = 0; index < length; index++) {
//            Node node = list.item( index );
//
//            // Much of this copied to Suspend.Suspend - refactor
//            if (null != node) {
//                if (node.getNodeType() == Node.ELEMENT_NODE) {
//                    Element element = (Element) node;
//                    new Event( element );
//                }
//            }
//        }
//    }

    public static String Name() throws ReferenceException {
        if (null != Only) {
            return Only.id;
        }

        throw new ReferenceException( "Event is not defined." );
    }

    public Event( Element element ) throws AttributeMissingException, InvalidXMLException {
        super( element );

        id = getStringAttribute( element, "name" );

        Only = this;
    }

    @Override
    public void verify() throws FeatureException, InvalidXMLException, LocationException, ReferenceException {
    }

    @Override
    public void dom( Draw draw, View mode ) throws MountingException, ReferenceException {

        StringBuilder title = new StringBuilder( id + " - " );

        switch (mode) {
            case PLAN:
                title.append( "Plan view" );
                break;
            case SECTION:
                title.append( "Section view" );
                break;
            case FRONT:
                title.append( "Front view" );
                break;

        }
        draw.setDocumentTitle( title.toString() );
    }
}

File: src/com/mobiletheatertech/plot/Event.scala
package com.mobiletheatertech.plot

import org.w3c.dom.Element

/**
  * Created by DHS on 8/2/16.
  */
class Event ( element: Element ) extends UniqueId ( element )
  with Populate
{
//  println( "Event")

  Event.Only = this

  tagCallback( PipeBase.Tag, processPipeBase )
  tagCallback( TrussBase.Tag, processTrussBase )
  tagCallback( Truss.Tag, processTruss )
  tagCallback( SetPiece.Tag, processSetPiece )
  tagCallback( Luminaire.LAYERTAG, processLuminaire )
//  println( "Event populating...")
  populate( element )
//  println( "Event done populating.")

  def processPipeBase( element: Element ): Unit = {
    new PipeBase( element )
  }

  def processTrussBase( element: Element ): Unit = {
//    println( "Event about to create TrussBase")
    new TrussBase( element )
  }

  def processTruss( element: Element ): Unit = {
//    println( "Event about to create Truss")
    new Truss( element )
//    println( "Event completed creating Truss")
  }

  def processSetPiece( element: Element ): Unit = {
    new SetPiece( element )
  }

  def processLuminaire( element: Element ): Unit = {
    new Luminaire( element )
  }

  def dom( draw: Draw, mode: View ): Unit = {

    val title = new StringBuilder( id )

    mode match {
      case View.PLAN => title.append( " - Plan view" )
      //    case View.TRUSS => return
      //    case View.SECTION => title.append( "Section view" )
      //    case View.FRONT => title.append( "Front view" )
      case _ => ;
    }

    draw.setDocumentTitle( title.toString() )
  }

  override def verify(): Unit = {
  }
}

object Event {
  final val Tag: String = "event"

  var Only: Event = null

  def Name: String = {
    if ( null == Only ) {
      throw new InvalidXMLException( "Event is not defined." )
    }
    Only.id
  }

  def Reset: Unit = {
    Only = null
  }
}
